test/client.spec.ts

```diff
@@ -62,10 +63,10 @@ describe('client', () => {
           }
         };
 
-        fetchMock.mock('end:test_config', successResponse, options)
+        fetchMock.mock(`${hostname}/rest/v1.1/test_config`, successResponse, options)
```

Comment:
This test called `'allows headers override'` and the purpose of particular this test is to validate headers override logic.

It might be better, do not to touch original tests, but to implement a separate tests suite with test target to validate the new 'apiVersion' optional parameter.

P.S.  I'll merge this branch (it's extremely helpful), and do refactoring for the existing tests and write a few new ones.
@gilmcquillan  Many thanks for the help.